fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1384,4 +1386,242 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
+                                .append(JOIN + resourceType.getSimpleName() + "_RESOURCES AS " + prevChainedResourceVar)
+                                .append(ON + prevChainedResourceVar + DOT + "RESOURCE_ID" + EQ + prevChainedLogicalResourceVar + DOT + "CURRENT_RESOURCE_ID")
+                                .append(AND + prevChainedResourceVar + DOT + "IS_DELETED" + EQ + "'N'")
+                                .append(WHERE);
+            }
+
+            if (parmIndex < lastParmIndex) {
+                if (currentParm.isReverseChained()) {
+                    // Build inner select joins: EXISTS (SELECT 1
+                    //                           FROM <modifierTypeResourceName>_TOKEN_VALUES_V AS CPx
+                    //                           JOIN <modifierTypeResourceName>_LOGICAL_RESOURCES AS CLRx
+                    //                             ON CLRx.LOGICAL_RESOURCE_ID = CPx.LOGICAL_RESOURCE_ID
+                    //                           JOIN <modifierTypeResourceName>_RESOURCES AS CRx
+                    //                             ON CRx.RESOURCE_ID = CLRx.CURRENT_RESOURCE_ID AND CRx.IS_DELETED = 'N'
+                    selectSegments.append("EXISTS " + LEFT_PAREN + "SELECT 1")
+                                    .append(FROM + currentParm.getModifierResourceTypeName() + "_TOKEN_VALUES_V AS " + chainedParmVar)
+                                    .append(JOIN + currentParm.getModifierResourceTypeName() + "_LOGICAL_RESOURCES AS " + chainedLogicalResourceVar)
+                                    .append(ON + chainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID" + EQ + chainedParmVar + DOT + "LOGICAL_RESOURCE_ID")
+                                    .append(JOIN + currentParm.getModifierResourceTypeName() + "_RESOURCES AS " + chainedResourceVar)
+                                    .append(ON + chainedResourceVar + DOT + "RESOURCE_ID" + EQ + chainedLogicalResourceVar + DOT + "CURRENT_RESOURCE_ID")
+                                    .append(AND + chainedResourceVar + DOT + "IS_DELETED" + EQ + "'N'");
+
+                    String referencedResourceType = null;
+                    if (parmIndex == 0) {
+                        referencedResourceType = resourceType.getSimpleName();
+                    } else {
+                        referencedResourceType = previousParm.getModifierResourceTypeName();
+                    }
+                    if (parmIndex < lastParmIndex - 1 && currentParm.getNextParameter().isReverseChained()) {
+                        // Build inner select where clause: WHERE CPx.TOKEN_VALUE = CLR<x-1>.LOGICAL_ID
+                        //                                    AND CPx.PARAMETER_NAME_ID = <parm-name-id>
+                        //                                    AND CPx.CDE_SYSTEM_ID = <code-system-id>
+                        //                                    AND
```

Comment:
Where you have the nice format, we can add the @formatter:on

Reply:
added formatter annotations